modules/reindex/src/test/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSourceTests.java

```diff
@@ -433,39 +429,6 @@ public void testWrapExceptionToPreserveStatus() throws IOException {
         assertEquals(badEntityException, wrapped.getSuppressed()[0]);
     }
 
-    @SuppressWarnings({ "unchecked", "rawtypes" })
-    public void testTooLargeResponse() throws Exception {
-        ContentTooLongException tooLong = new ContentTooLongException("too long!");
-        CloseableHttpAsyncClient httpClient = mock(CloseableHttpAsyncClient.class);
-        when(httpClient.<HttpResponse>execute(any(HttpAsyncRequestProducer.class), any(HttpAsyncResponseConsumer.class),
-                any(FutureCallback.class))).then(new Answer<Future<HttpResponse>>() {
-                    @Override
-                    public Future<HttpResponse> answer(InvocationOnMock invocationOnMock) throws Throwable {
-                        HeapBufferedAsyncResponseConsumer consumer = (HeapBufferedAsyncResponseConsumer) invocationOnMock.getArguments()[1];
-                        FutureCallback callback = (FutureCallback) invocationOnMock.getArguments()[2];
-
-                        assertEquals(new ByteSizeValue(200, ByteSizeUnit.MB).bytesAsInt(), consumer.getBufferLimit());
-                        callback.failed(tooLong);
-                        return null;
-                    }
-                });
-        RemoteScrollableHitSource source = sourceWithMockedClient(true, httpClient);
-
-        AtomicBoolean called = new AtomicBoolean();
-        Consumer<Response> checkResponse = r -> called.set(true);
-        Throwable e = expectThrows(RuntimeException.class,
-                () -> source.doStartNextScroll(FAKE_SCROLL_ID, timeValueMillis(0), checkResponse));
-        // Unwrap the some artifacts from the test
-        while (e.getMessage().equals("failed")) {
-            e = e.getCause();
-        }
-        // This next exception is what the user sees
-        assertEquals("Remote responded with a chunk that was too large. Use a smaller batch size.", e.getMessage());
```

Comment:
I'd like to keep this somehow so that we continue to test that we give users a good error message when they blow out the buffer.